recipes/binutils/all/conanfile.py

```diff
@@ -126,64 +129,78 @@ def _raise_unsupported_configuration(self, key, value):
-                  strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self.source_folder)
 
     @property
     def _exec_prefix(self):
         return os.path.join(self.package_folder, "bin", "exec_prefix")
 
-    @functools.lru_cache(1)
-    def _configure_autotools(self):
-        autotools = AutoToolsBuildEnvironment(self, win_bash=self._settings_build.os == "Windows")
+    def generate(self):
         yes_no = lambda tf : "yes" if tf else "no"
-        conf_args = [
-            f"--target={self.options.target_triplet}",
-            f"--enable-multilib={yes_no(self.options.multilib)}",
-            "--with-system-zlib",
-            "--disable-nls",
-            f"--program-prefix={self.options.prefix}",
-            f"exec_prefix={tools.unix_path(self._exec_prefix)}",
-        ]
-        autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return autotools
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.append("--disable-nls")
+        tc.configure_args.append(f"--target={self.options.target_triplet}")
+        tc.configure_args.append(f"--enable-multilib={yes_no(self.options.multilib)}")
+        tc.configure_args.append(f"--with-zlib={self.deps_cpp_info['zlib'].rootpath}")
+        tc.configure_args.append(f"--program-prefix={self.options.prefix}")
+        tc.configure_args.append(f"--exec_prefix={unix_path(self, os.path.join('/','bin', 'exec_prefix'))}")
+        tc.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        autotools = self._configure_autotools()
+        for _patch in self.conan_data.get("patches", {}).get(self.version, []):
+            patch(self, **_patch)
+        autotools = Autotools(self)
+        autotools.configure()
         autotools.make()
 
     def package(self):
-        self.copy("COPYING*", src=self._source_subfolder, dst="licenses")
-        autotools = self._configure_autotools()
+        autotools = Autotools(self)
         autotools.install()
 
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"), recursive=True)
+        copy(
+            self,
+            pattern="COPYING*",
+            dst=os.path.join(self.package_folder, "licenses"),
+            src=self.source_folder,
+            keep_path=False,
+        )
 
     def package_info(self):
+        target_bindir = os.path.join(self._exec_prefix, str(self.options.target_triplet), "bin")
+        self.output.info("Prepending PATH environment variable: {}".format(target_bindir))
+        self.buildenv_info.prepend_path("PATH", target_bindir)
```

Comment:
Not required - it's added by default in the new generator
```suggestion
```

Reply:
This isn't the `bin` directory in the package folder. Relative to the package folder, this would be `{package_folder}/bin/exec_prefix/x86_64-pc-linux-gnu/bin`, so not automatically added to the path to my understanding.

Reply:
you should add this path to cpp_info.bindirs